### DISCUSSION TITLE
Anyone else with Vite HMR port issues?

### DISCUSSION THREAD

**Discussion by technicallypete** [nuxt/nuxt#17580] [Questions]
With a standalone vite+vue setup, vite options for changing the HMR port works fine.

With nuxt3 though, it just "wont" work changing the port and I have to use the default 24678 ws port.

The config seems to work where it changes the port in the client, but it cant connect (server not running at the port.)

Anyone else experience this and/or got it to work? I don't think keeping it at 24678 is bad, but you should be able to specify a different port if needed (working on other projects locally at the same time for example, etc.)

**Comment by zangrus** [comment]
Hi @peteromano,
I spent several days to investigate this issue, and decided to give my response even though you asked the question six months ago and probably found already a solution. I hope it could be useful for others. So, if you want to change the port, say, to 7777, just put in the nuxt.config.ts:

```
export default defineNuxtConfig({
    vite: {
        server: {
            hmr: {
                clientPort: 7777, // the port to be sent to client
                port: 7777          // the port ws server listen to
            },
        },
    }
})
```
